### PR TITLE
Update: ARIA elements with only presentational descendants

### PR DESCRIPTION
**Pull request by estelle**
- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [X] Fixes a typo, bug, or other error

There are some types of user interface components that, when represented in a platform accessibility API, can only contain text. For example, accessibility APIs do not have a way of representing semantic elements contained in a button. To deal with this limitation, WAI-ARIA requires browsers to automatically apply role presentation to all descendant elements of any element with a role that cannot support semantic children.

The roles that require all children to be presentational are:

button
checkbox
img
meter
menuitemcheckbox
menuitemradio
option
progressbar
radio
scrollbar
separator
slider
switch
tab

This PR adds that information to (most of) the above roles.

menuitemcheckbox, menuitemradio, and scrollbar are not included in this PR as they are currently being worked on under other PRs: https://github.com/openwebdocs/project/issues/82

### CODE REVIEW

File: files/en-us/web/accessibility/aria/roles/img_role/index.md [hunk 1]
```diff
@@ -89,6 +89,28 @@ Another example where this might be suitable is when using ASCII emoji combinati
 
 If `aria-labelledby` were used, the screen reader would read it. In this case, only the contents of the `aria-label` are announced to screen reader users, hiding the gibberish of the characters without the need for descendant ARIA to hide things, but also hiding potential content that may be part of the image.
 
+### All descendants are presentational
+
+There are some types of user interface components that, when represented in a platform accessibility API, can only contain text. Accessibility APIs do not have a way of representing semantic elements contained in an `img`. To deal with this limitation, browsers, automatically apply role [`presentation`](/en-US/docs/Web/Accessibility/ARIA/Roles/presentation_role) to all descendant elements of any `img` element as it is a role that does not support semantic children.
+
+For example, consider the following ```` element, which contains a heading.
```
Comment (estelle) [inline files/en-us/web/accessibility/aria/roles/img_role/index.md:96]:
```suggestion
For example, consider the following `img` element, which contains a heading.
```

File: files/en-us/web/accessibility/aria/roles/meter_role/index.md [hunk 2]
```diff
@@ -20,6 +20,28 @@ Each element with `role="meter"` must also have one of the following:
 - An [`aria-label`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) attribute.
 - An [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) attribute pointing to an element with text that describes the meter.
 
+### All descendants are presentational
+
+There are some types of user interface components that, when represented in a platform accessibility API, can only contain text. Accessibility APIs do not have a way of representing semantic elements contained in a `meter`. To deal with this limitation, browsers, automatically apply role [`presentation`](/en-US/docs/Web/Accessibility/ARIA/Roles/presentation_role) to all descendant elements of any `meter` element as it is a role that does not support semantic children.
+
+For example, consider the following ```` element, which contains a heading.
```
Comment (estelle) [inline files/en-us/web/accessibility/aria/roles/meter_role/index.md:27]:
```suggestion
For example, consider the following `meter` element, which contains a heading.
```

File: files/en-us/web/accessibility/aria/roles/option_role/index.md [hunk 3]
```diff
@@ -23,6 +23,28 @@ Authors can also explicitly provide an accessible name by specifying [`aria-labe
 
 It is highly recommended to use a {{HTMLElement('select')}} element or an  {{HTMLElement('input')}} element with the `checkbox` or `radio` type instead, when possible. These native HTML elements provide keyboard interactivity to manage focus for all the descendants for you automatically.
 
+### All descendants are presentational
+
+There are some types of user interface components that, when represented in a platform accessibility API, can only contain text. Accessibility APIs do not have a way of representing semantic elements contained in a `option`. To deal with this limitation, browsers, automatically apply role [`presentation`](/en-US/docs/Web/Accessibility/ARIA/Roles/presentation_role) to all descendant elements of any `option` element as it is a role that does not support semantic children.
+
+For example, consider the following ```` element, which contains a heading.
```
Comment (estelle) [inline files/en-us/web/accessibility/aria/roles/option_role/index.md:30]:
```suggestion
For example, consider the following `option` element, which contains a heading.
```

File: files/en-us/web/accessibility/aria/roles/progressbar_role/index.md [hunk 4]
```diff
@@ -27,6 +27,28 @@ Authors **may** set aria-valuemin and aria-valuemax to indicate the minimum and
 
 If the `progressbar` role is applied to an HTML {{HTMLElement('progress')}} element, the accessible name can come from the associated {{HTMLElement('label')}}. Otherwise use [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) if a visible label is present or  [`aria-label`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) if a visible label is not present.
 
+### All descendants are presentational
+
+There are some types of user interface components that, when represented in a platform accessibility API, can only contain text. Accessibility APIs do not have a way of representing semantic elements contained in a `progressbar`. To deal with this limitation, browsers, automatically apply role [`presentation`](/en-US/docs/Web/Accessibility/ARIA/Roles/presentation_role) to all descendant elements of any `progressbar` element as it is a role that does not support semantic children.
+
+For example, consider the following ```` element, which contains a heading.
```
Comment (estelle) [inline files/en-us/web/accessibility/aria/roles/progressbar_role/index.md:34]:
```suggestion
For example, consider the following `progressbar` element, which contains a heading.
```

File: files/en-us/web/accessibility/aria/roles/radio_role/index.md [hunk 5]
```diff
@@ -72,6 +72,28 @@ As `radio` is an interactive control; it must be focusable and keyboard accessib
 
 If any of the radio roles in a group has `aria-required="true"` set, it is as if all of the radios in the group had the attribute making the selection of one of the radios in the radiogroup being required to be valid;  though not necessarily the one that has the [`aria-required`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-required) attribute set.
 
+### All descendants are presentational
+
+There are some types of user interface components that, when represented in a platform accessibility API, can only contain text. Accessibility APIs do not have a way of representing semantic elements contained in a `radio`. To deal with this limitation, browsers, automatically apply role [`presentation`](/en-US/docs/Web/Accessibility/ARIA/Roles/presentation_role) to all descendant elements of any `radio` element as it is a role that does not support semantic children.
+
+For example, consider the following ```` element, which contains a heading.
```
Comment (estelle) [inline files/en-us/web/accessibility/aria/roles/radio_role/index.md:79]:
```suggestion
For example, consider the following `radio` element, which contains a heading.
```

File: files/en-us/web/accessibility/aria/roles/separator_role/index.md [hunk 6]
```diff
@@ -63,6 +63,28 @@ If the separator is focusable, providing a visible boundary between two sections
 
 An accessible name, with [`aria-label`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) should be included if there is more than one focusable separator.
 
+### All descendants are presentational
+
+There are some types of user interface components that, when represented in a platform accessibility API, can only contain text. Accessibility APIs do not have a way of representing semantic elements contained in a `separator`. To deal with this limitation, browsers, automatically apply role [`presentation`](/en-US/docs/Web/Accessibility/ARIA/Roles/presentation_role) to all descendant elements of any `separator` element as it is a role that does not support semantic children.
+
+For example, consider the following ```` element, which contains a heading.
```
Comment (estelle) [inline files/en-us/web/accessibility/aria/roles/separator_role/index.md:70]:
```suggestion
For example, consider the following `separator` element, which contains a heading.
```

File: files/en-us/web/accessibility/aria/roles/slider_role/index.md [hunk 7]
```diff
@@ -70,6 +70,28 @@ In many two-thumb sliders, the thumbs are not allowed to pass one another, such
 
 It is not a requirement that the thumbs in multi-thumb sliders be dependent on the other thumb values, but intuitive user experience is a requirement, so it is recommended to avoid this anti-pattern.
 
+### All descendants are presentational
+
+There are some types of user interface components that, when represented in a platform accessibility API, can only contain text. Accessibility APIs do not have a way of representing semantic elements contained in a `slider`. To deal with this limitation, browsers, automatically apply role [`presentation`](/en-US/docs/Web/Accessibility/ARIA/Roles/presentation_role) to all descendant elements of any `slider` element as it is a role that does not support semantic children.
+
+For example, consider the following ```` element, which contains a heading.
```
Comment (estelle) [inline files/en-us/web/accessibility/aria/roles/slider_role/index.md:77]:
```suggestion
For example, consider the following `slider` element, which contains a heading.
```

File: files/en-us/web/accessibility/aria/roles/switch_role/index.md [hunk 8]
```diff
@@ -32,6 +32,28 @@ Assistive technologies may choose to represent `switch` widgets with a specializ
 
 Since a switch is an interactive control, it must be focusable and keyboard accessible. If the role is applied to a non-focusable element, use the `tabindex` attribute to change this. The expected keyboard shortcut for toggling the value of a switch is the <kbd>Space</kbd> key. The developer is required to change the value of the `aria-checked` attribute dynamically when the switch is toggled.
 
+### All descendants are presentational
+
+There are some types of user interface components that, when represented in a platform accessibility API, can only contain text. Accessibility APIs do not have a way of representing semantic elements contained in a `switch`. To deal with this limitation, browsers, automatically apply role [`presentation`](/en-US/docs/Web/Accessibility/ARIA/Roles/presentation_role) to all descendant elements of any `switch` element as it is a role that does not support semantic children.
+
+For example, consider the following ```` element, which contains a heading.
```
Comment (estelle) [inline files/en-us/web/accessibility/aria/roles/switch_role/index.md:39]:
```suggestion
For example, consider the following `switch` element, which contains a heading.
```

File: files/en-us/web/accessibility/aria/roles/tab_role/index.md [hunk 9]
```diff
@@ -22,6 +22,28 @@ Elements with the role `tab` *must* either be a child of an element with the `ta
 
 When elements with the `tab` role are selected or active they should have their [`aria-selected`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-selected) attribute set to `true`. Otherwise, their `aria-selected` attribute should be set to `false`. When a `tab` is selected or active, its corresponding controlled `tabpanel` should have its [`aria-expanded`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded) attribute set to `true` and its `hidden` attribute set to `false`, otherwise the reverse.
 
+### All descendants are presentational
+
+There are some types of user interface components that, when represented in a platform accessibility API, can only contain text. Accessibility APIs do not have a way of representing semantic elements contained in a `tab`. To deal with this limitation, browsers, automatically apply role [`presentation`](/en-US/docs/Web/Accessibility/ARIA/Roles/presentation_role) to all descendant elements of any `tab` element as it is a role that does not support semantic children.
+
+For example, consider the following ```` element, which contains a heading.
```
Comment (estelle) [inline files/en-us/web/accessibility/aria/roles/tab_role/index.md:29]:
```suggestion
For example, consider the following `tab` element, which contains a heading.
```

**Review by ericwbailey — APPROVED** [review]
Well, this is lovely. Thank you 🙂

**Pull request merged**